### PR TITLE
[SPARKR-156] phase 2: implement zipWithIndex() of the RDD class.

### PR DESCRIPTION
**Pull request by sun-rui**

### CODE REVIEW

File: pkg/R/RDD.R [hunk 1]
```diff
@@ -1468,6 +1468,64 @@ setMethod("zipWithUniqueId",
...
           })
 
+#' Zip an RDD with its element indices.
+#'
+#' The ordering is first based on the partition index and then the
+#' ordering of items within each partition. So the first item in
+#' the first partition gets index 0, and the last item in the last
+#' partition receives the largest index.
+#'
+#' This method needs to trigger a spark job when this RDD contains
```
Comment (concretevitamin) [inline pkg/R/RDD.R]: nit: s/spark/Spark; also "one partition" (singular)

**Comment by concretevitamin** [comment]
LGTM pending a minor grammar comment

**Comment by sun-rui** [comment]
Fix comments.

File: pkg/R/RDD.R (hunk 1 above)
Comment (sun-rui) [inline pkg/R/RDD.R]: fixed

**Comment by shivaram** [comment]
Thanks @sun-rui . Merging this.

**Pull request merged**